src/librustc_trans/trans/consts.rs

```diff
@@ -130,13 +132,14 @@ fn addr_of_mut(ccx: &CrateContext,
 
 pub fn addr_of(ccx: &CrateContext,
                cv: ValueRef,
+               align: machine::llalign,
                kind: &str)
                -> ValueRef {
     match ccx.const_globals().borrow().get(&cv) {
         Some(&gv) => return gv,
```

Comment:
No, not just an assert: you can legitimately end up with a value with the wrong alignment.  Consider:

```rust
#![crate_type="lib"]
#[repr(i16)]
pub enum E<A, B> {
    A(A),
    B(B),
}

pub fn f() -> E<i16, [i16; 3]> {
    E::A(0)
}

pub fn g() -> E<i16, i32> {
     E::A(0)
}
```